pkg/kubectl/cmd/run.go

```diff
@@ -304,45 +350,98 @@ func contains(resourcesList map[string]*unversioned.APIResourceList, resource un
 	return false
 }
 
-func waitForPodRunning(c *client.Client, pod *api.Pod, out io.Writer, quiet bool) (status api.PodPhase, err error) {
+func waitForPod(c *client.Client, ns, name string, exitCondition func(*api.Pod) bool, tick func(*api.Pod)) (*api.Pod, error) {
```

Comment:
``tick`` argument together with semantic of ``waitForPod`` awakens interpretation of ``tick`` to be the ticker function controlling the for loop inside. What about to change ``tick`` to ``notify``?